TheraLang.BLL/Services/ProjectParticipationService.cs

```diff
@@ -25,35 +25,35 @@ public async Task ChangeStatusAsync(int participantId, ProjectParticipationStatu
         {
             try
             {
-                ProjectParticipation participant = _unitOfWork.Repository<ProjectParticipation>()
-                                                              .Get().SingleOrDefault(x => x.Id == participantId);
+                ProjectParticipation participant = await _unitOfWork.Repository<ProjectParticipation>()
+                    .GetAsync(x => x.Id == participantId);
 
                 if (participant == null)
                 {
-                    throw new NullReferenceException($"Error while changing status. Participant with id {nameof(participantId)}={participantId} not found");
+                    throw new NullReferenceException(
+                        $"Error while changing status. Participant with id {nameof(participantId)}={participantId} not found");
                 }
 
-                var mapper = new MapperConfiguration(cfg => cfg.CreateMap<ProjectParticipationStatusDto, ProjectParticipationStatus>()).CreateMapper();
+                var mapper = new MapperConfiguration(cfg =>
+                    cfg.CreateMap<ProjectParticipationStatusDto, ProjectParticipationStatus>()).CreateMapper();
                 var status = mapper.Map<ProjectParticipationStatusDto, ProjectParticipationStatus>(statusDto);
 
                 participant.Status = status;
 
-                _unitOfWork.Repository<ProjectParticipation>().Update(participant);
+                await _unitOfWork.Repository<ProjectParticipation>().UpdateAsync(participant);
                 await _unitOfWork.SaveChangesAsync();
             }
             catch (Exception ex)
             {
                 throw new Exception($"Error while updating the status for {nameof(participantId)}:{participantId}", ex);
             }
-
         }
 
-        public IEnumerable<ProjectParticipationDto> GetAll()
+        public async Task<IEnumerable<ProjectParticipationDto>> GetAllAsync()
         {
-            var projectParticipations = _unitOfWork.Repository<ProjectParticipation>().Get()
-                .Include(p => p.User)
-                .Include(p => p.Project)
-                .ToList();
+            var projectParticipations = await _unitOfWork.Repository<ProjectParticipation>()
+                .GetListWithIncludeAsync("User", "Project");
```

Comment:
don't really like this new implementation, i would prefer linq over string params passing